pkg/ccpatch/patch_userdata.go

```diff
@@ -58,6 +108,27 @@ func (c *CloudInitPatcher) Patch(key string, value any) *CloudInitPatcher {
 	return c
 }
 
+// SetEnv sets an environment variable in the cloud-init data
+// within the "environment" key
+func (c *CloudInitPatcher) SetEnv(key string, value string) *CloudInitPatcher {
+	if c.data["environment"] == nil {
+		c.data["environment"] = make(map[string]interface{})
+	}
+
+	c.data["environment"].(map[string]interface{})[key] = value
+
+	return c
+}
+
+// GetEnv returns the value of an environment variable in the cloud-init data
+func (c *CloudInitPatcher) GetEnv(key string) string {
+	if c.data["environment"] == nil {
+		return ""
+	}
+
+	return c.data["environment"].(map[string]interface{})[key].(string)
+}
+
```

Comment:
They are doing completly different things

```
return c.data["environment"].(map[string]interface{})[key].(string)
```

c.data[environment] has been casted/converted to a map[string]interface{} so we can access `key`, The value of `key` gets casted to .(string)

https://go.dev/tour/basics/13

Reply:
tbh: yes, this would be good practice, but if we start to optimize all of our allocations, this would result in a lot rewriting for the entire operator:

```go
for _, value := x {
     fmt.Println(value)
}
```

vs 
```go
for i := range x {
    fmt.Println(x[i])
}
```

so i would take it as it is.

Reply:
Ok then let's leave it as it is